vm/actor/src/builtin/market/mod.rs

```diff
@@ -229,9 +218,19 @@ impl Actor {
             );
         }
 
-        let (_, worker, _) = request_miner_control_addrs(rt, provider)?;
-        if &worker != rt.message().caller() {
-            return Err(actor_error!(ErrForbidden; "Caller is not provider {}", worker));
+        let (_, worker, controllers) = request_miner_control_addrs(rt, provider)?;
+        let caller = rt.message().caller();
+        let mut caller_ok = caller == &worker;
+        for controller in controllers.iter() {
+            if caller_ok {
+                break;
+            }
+            caller_ok = caller == controller;
+        }
+        if !caller_ok {
+            return Err(
+                actor_error!(ErrForbidden; "caller {} is now worker or control address of provider {}", caller, provider),
```

Comment:
No need to change here, but you'll run into less formatting issues later on if you use the `,` separator instead of `;`. For some reason semicolon in a macro avoids formatting and can void formatting for the whole file if over the width limit. I had been slowly transitioning over with the plan of removing support for the `;`, but totally optional.